examples/ex32p_proposed.cpp

```diff
@@ -0,0 +1,455 @@
+//                       MFEM Example 32 - Parallel Version
+//
+// Compile with: make ex32p
+//
+// Sample runs:  mpirun -np 4 ex32p -m ../data/square-disc.mesh -o 1
```

Comment:
More than one of sample runs have visualization problems, similar to the case reported above, when they are run with MPI.

It appears that all of the sample runs work when run in serial. However, some of the runs deliver very suspicious refinement patterns. It is possible that the error estimator being used here is not appropriate "right out of the box" for practical applications. Are you sure that this example should be merged into the main code? Another option is stating somewhere that there is "experimental support" for AMR/a posteriori error estimation.

Reply:
Yes, I noticed this a few days ago and I've been trying to track down the problem (without much luck). I may pull `ex32p_proposed ` out of this PR. Sorry for the hassle.

Reply:
> Yes, I noticed this a few days ago and I've been trying to track down the problem (without much luck). I may pull `ex32p_proposed ` out of this PR. Sorry for the hassle.

I was playing with this too, just now, and what happens in parallel runs is that the solver produces NANs, in a non-deterministic way. I noticed that the source of this is the call to 
```cpp
sol.ProjectBdrCoefficientTangent(E, ess_bdr);
````
If replaced with `sol.ProjectCoefficient(E);` the non-deterministic behavior goes away. There is still a Bad termination at the 17th AMR iteration but it's deterministic and up to that point everything looks good.

Reply:
This sounds like a useful clue. Thanks, @psocratis!

Ever since these `ProjectBdr*` methods started using `AccumulateAndCount*` they keep causing me trouble. We should see if we have any unit tests that cover these. Not in this PR but soon...

Reply:
I created an issue (#2226) so that we don't forget to resolve this.